src/sentry/api/endpoints/api_tokens.py

```diff
@@ -83,27 +103,21 @@ def post(self, request: Request) -> Response:
 
     @method_decorator(never_cache)
     def delete(self, request: Request):
-        user_id = request.user.id
-        if is_active_superuser(request):
-            user_id = request.data.get("userId", user_id)
-        # TODO: we should not be requiring full token value in the delete endpoint, and should instead be using the id
-        token = request.data.get("token", None)
+        user_id = self._get_appropriate_user_id(request=request)
         token_id = request.data.get("tokenId", None)
         # Account for token_id being 0, which can be considered valid
-        if not token and token_id is None:
-            return Response({"token": token, "tokenId": token_id}, status=400)
+        if token_id is None:
+            return Response({"tokenId": token_id}, status=400)
 
         with outbox_context(transaction.atomic(router.db_for_write(ApiToken)), flush=False):
-            if token:
-                for token in ApiToken.objects.filter(
-                    user_id=user_id, token=token, application__isnull=True
-                ):
-                    token.delete()
-            else:
-                token_to_delete = ApiToken.objects.get(
-                    id=token_id, application__isnull=True, user_id=user_id
-                )
-                token_to_delete.delete()
+            token_to_delete: Optional[ApiToken] = ApiToken.objects.filter(
```

Comment:
No chances that we have multiple tokens returned here is there?
`first()` has me a li'l sus 🤔

Reply:
Since we're using the `id` we'll only ever get 1, but this is a way to not have a try/catch in this area. I can utilize `.get` which would side step the `filter().first()`, but then I need to make sure I catch the exception for does not exist. 

https://docs.djangoproject.com/en/5.0/ref/models/querysets/#get

https://docs.djangoproject.com/en/5.0/ref/models/querysets/#first